src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3147,6 +3146,11 @@ protected virtual void Dispose(bool disposing)
                     if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, "Calling _handle.Dispose()");
                     handle.Dispose();
                 }
+                else if (!handle.OwnsHandle)
```

Comment:
nit: I personally don't like the negated booleans. I would use `if (handle.OwnsHandle))` for the original block and add `CloseAsIs()` as new else.